garrysmod/lua/includes/modules/draw.lua

```diff
@@ -132,15 +134,15 @@ function MultiColorText( font, x, y, xAlign, yAlign, ... )
 	local w, h = 0, 0
 
 	if ( xAlign != TEXT_ALIGN_LEFT ) then
-		local fullText = ""
-		
+		table.Empty( MultiColorText_StringTable )
+
 		for k, v in ipairs, table, origin do
 			if not IsColor( v ) then
-				fullText = fullText .. tostring( v )
+				table.insert( MultiColorText_StringTable, tostring( v ) )

```

Comment:
the best way to do this if you started out is making a local `i` and using it instead of `tbl[#tbl + 1] = `

Reply:
Ok I can't tell if I'm stupid or not because I have no idea what you mean.

Reply:
Make a local and count it in the loop separately